### PR TITLE
Numerous updates

### PR DESCRIPTION
**Pull request by treydock**
* Require Puppet 7 or Puppet 8
* Support stdlib 9
* Support EL9
* Lint fixes

Unit tests worked:

```
Finished in 19.31 seconds (files took 1.56 seconds to load)
14 examples, 0 failures
```

### CODE REVIEW

File: data/RedHat-9.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Specific options for RedHat 8 systems
```
Comment (joshbeard) [inline data/RedHat-9.yaml]: s/8/9

**Comment by joshbeard** [comment]
Hey @treydock, thanks for this contribution and much needed update! Will get this merged and published ASAP - one minor tweak to the comment, though, if you don't mind.

File: data/RedHat-9.yaml (hunk 1 above)
Comment (treydock) [reply data/RedHat-9.yaml]: Good catch, this is fixed.

Comment (joshbeard) [reply data/RedHat-9.yaml]: Thank you! and thanks again for the updates.

**Review by joshbeard — APPROVED** [review]

**Comment by joshbeard** [comment]
This is published to the Puppet Forge in version 1.2.0: https://forge.puppet.com/modules/joshbeard/login_defs

**Pull request merged**